### PR TITLE
[fix](predicate) fix be core dump caused by pushing down the double column predicate

### LINKED ISSUE #15678: [Bug] [daily] be core when execute split_part query
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master doris-0.0.0-trunk-95f2f43

### What's Wrong?
jenkins failed case: https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/167/testReport/(root)/test_query_string_function/test_query_split_part/

image
image

### What You Expected?

query ok

### How to Reproduce?

select split_part(k8, '1', 1), k8, split_part(concat(k8, '12'), '1', 1) from baseall order by k8 limit 2;

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #15678

related pr: #15558

## Problem summary

The double column predicate is pushed down to the storage layer to cause a core dump, because there is no PredicateCreator corresponding to the double type. It looks like none of the double column predicates were pushed to the storage layer before, and I'm not quite sure why.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.07 seconds
 load time: 475 seconds
 storage size: 17121111998 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230107032959_clickbench_pr_75215.html

**Comment by Gabriel39** [comment]
PTAL @xiaokang

**Comment by xiaokang** [comment]
> PTAL @xiaokang

I will check it.

**Comment by xiaokang** [comment]
The root cause is that there is no implementation for float/doube in `get_creator` function in `be/src/olap/predicate_creator.h`. `get_creator` will return default value nullptr and cause create_predicate core in the following code.

> template <PredicateType PT, typename ConditionType>
inline ColumnPredicate* create_predicate(const TabletColumn& column, int index,
                                         const ConditionType& conditions, bool opposite,
                                         MemPool* pool) {
    return get_creator<PT, ConditionType>(column.type())
            ->create(column, index, conditions, opposite, pool);
}

**Comment by luozenglin** [comment]
> create_predicate

Yes, but what confuses me is why the `create_predicate` of type double was not implemented before? In other words, the double type was not pushed down to the storage layer before because of any pitfalls?

**Comment by luozenglin** [comment]
> The root cause is that there is no implementation for float/doube in `get_creator` function in `be/src/olap/predicate_creator.h`. `get_creator` will return default value nullptr and cause create_predicate core in the following code.
> 
> > template <PredicateType PT, typename ConditionType>
> > inline ColumnPredicate* create_predicate(const TabletColumn& column, int index,
> > const ConditionType& conditions, bool opposite,
> > MemPool* pool) {
> > return get_creator<PT, ConditionType>(column.type())
> > ->create(column, index, conditions, opposite, pool);
> > }

Yes, but what confuses me is why the create_predicate of type double was not implemented before? In other words, the double type was not pushed down to the storage layer before because of any pitfalls?

**Comment by xiaokang** [comment]
> Yes, but what confuses me is why the create_predicate of type double was not implemented before? In other words, the double type was not pushed down to the storage layer before because of any pitfalls?

I have confirmed float/double type was not pushed down to the storage layer by checking the code in _normalize_conjuncts() and discussing with @moonming . The pitfalls may be the precision problem. I will do some research on other databases.

File: regression-test/suites/query_p0/sql_functions/string_functions/test_split_part.groovy [hunk 1]
```diff
@@ -27,4 +27,6 @@ suite("test_split_part") {
     """
     exception "[RUNTIME_ERROR]Argument at index 3 for function split_part must be constant"
   }
+
+  qt_1 "select split_part(k8, '1', 1), k8, split_part(concat(k8, '12'), '1', 1) from test_query_db.test order by k8 limit 2;"
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/string_functions/test_split_part.groovy:31]: Does this regression test related with the issue? It does not have any pushdown predicates.

Comment (luozenglin) [reply regression-test/suites/query_p0/sql_functions/string_functions/test_split_part.groovy:31]: topn generated a runtime predicate, a newly supported feature in pr #15558

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**